src/components/logview/logview.tsx

```diff
@@ -65,50 +64,24 @@ export class LogView {
   }
 
   transformToElement(data: LogData) {
-    switch (data.type) {
-      case LogDataType.Instruction:
-        return (
-          <div class="instruction entry">
-            <div class="title">{data.title}</div>
-            {data.body ? <div class="body">{data.body}</div> : null}
-          </div>
-        )
-
-      case LogDataType.Error:
-        return (
-          <div class="error entry">
-            <div class="title">{data.title}</div>
-            {data.body ? <div class="body">{data.body}</div> : null}
-          </div>
-        )
-
-      case LogDataType.Request:
-        const requestBody =
-          typeof data.body == 'string'
-            ? data.body
-            : JSON.stringify(data.body, null, 2)
-        return (
-          <div class="request entry">
-            <div class="title">{data.url}</div>
-            <div class="body">{requestBody}</div>
-          </div>
-        )
-
-      case LogDataType.Response:
-        const responseBody =
-          typeof data.body == 'string'
-            ? data.body
-            : JSON.stringify(data.body, null, 2)
-        return (
-          <div class="response entry">
-            <div class="title">{data.url}</div>
-            <div class="body">{responseBody}</div>
-          </div>
-        )
-
-      default:
-        return null
-    }
+    const className = {
+      [LogDataType.Instruction]: 'instruction',
+      [LogDataType.Error]: 'error',
+      [LogDataType.Request]: 'request',
+      [LogDataType.Response]: 'response',
+    }[data.type]
```

Comment:
Won't this be a KeyError since the keys are wrapped in lists?

Reply:
This syntax is a way of using variables as the keys for objects

```
const a = "Hello";
const obj1 = {
  a: "World"
}
const obj2 = {
  [a]: "World"
}
```
Obj1 looks like `{a: "World"}` while obj2 looks like `{Hello: "World"}`